core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -258,9 +276,14 @@ class ReplicaFetcherThread(name: String,
       }
     }
 
-    val requestBuilder = JFetchRequest.Builder.forReplica(fetchRequestVersion, replicaId, maxWait, minBytes, requestMap)
-      .setMaxBytes(maxBytes)
-    ResultWithPartitions(new FetchRequest(requestBuilder), partitionsWithError)
+    val fetchData = builder.build()
+    val requestBuilder = JFetchRequest.Builder.forReplica(fetchRequestVersion, replicaId,
+        maxWait, minBytes, fetchData.toSend()).setMaxBytes(maxBytes)
+    if (fetchMetadataSupported) {
```

Comment:
It's a convenience to avoid putting references to specific versions into the body of the code.  Happy to replace it with `brokerConfig.interBrokerProtocolVersion >= KAFKA_1_1_IV0` if that makes sens?